src/Parser.js

```diff
@@ -32,6 +34,9 @@ class Parser {
         if (this.constructor === Parser) {
             throw new Error("Cannot instantiate abstract class Plugin directly!");
         }
+        if (!options) return;
+        this.API = options.API;
+        this.logger = (this.API && this.API.getLogger()) || new NullLogger();
```

Comment:
Logger should not be instantiated in here. If this dependency is optional, instead of constructor injection we could use a setter injection `setLogger(/** @type {Logger} */ logger)` and define `logger` property as `Logger | undefined` so that implementation of `Parser` would know that it should call `logger?.error("some message")`.

Reply:
removed